#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  info [
    "TP temps reel INSA 2024-2025"
  ]
  title "TP temps reel 4IR"
  FPS 30
}
Viewpoint {
  orientation -0.24848844346372556 -0.0822967449470058 0.9651324982799524 3.760359778390886
  position 2.2042506888942515 1.7024281569740527 1.528175955689362
  ambientOcclusionRadius 12
  bloomThreshold -1
}
TexturedBackground {
}
RectangleArena {
  floorSize 2 2
  floorAppearance Parquetry {
    type "chequered"
    IBLStrength 12
  }
}
DEF GREEN_BOX Solid {
  translation 0.03 0.72 0.05
  rotation 2.3464099999741633e-06 -4.064099999955249e-06 0.9999999999889888 -1.047195307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 0.2189099510000001
        metalness 0
      }
      geometry DEF BOX0 Box {
        size 0.1 0.23 0.1
      }
    }
  ]
  name "green box"
  boundingObject Shape {
    appearance PBRAppearance {
      metalness 0
    }
    geometry USE BOX0
  }
}
DEF BLUE_BOX Solid {
  translation -0.95 0 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 0.519341125
        metalness 0
      }
      geometry DEF BOX1 Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "blue box"
  boundingObject Shape {
    appearance PBRAppearance {
      metalness 0
    }
    geometry USE BOX1
  }
}
DEF YELLOW_BOX Solid {
  translation 0.356586 0.414161 0.05
  rotation 4.73079e-08 9.58979e-09 -1 -0.3999953071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 0.03934117200000009
        metalness 0
      }
      geometry DEF BOX2 Box {
        size 0.3 0.05 0.1
      }
    }
  ]
  name "yellow box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX2
  }
}
DEF PINK_BOX Solid {
  translation 0.95 0 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.1 0.3
        roughness 0.16827074099999995
        metalness 0
      }
      geometry DEF BOX3 Box {
        size 0.08 0.15 0.1
      }
    }
  ]
  name "red box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX3
  }
}
Robot {
  children [
    InertialUnit {
    }
    Pose {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1.7855312599102715e-07
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.045 0.025
        rotation 0.13448686832386902 -0.13448686832384543 -0.9817466905963487 1.589216975986931
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.785497881853092e-07
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.045 0.025
        rotation -0.0623569123629744 0.06235691236298219 -0.9961040261745315 1.5746999888661786
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499205
      children [
        DEF SENSOR Pose {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 1 -0.500795
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Pose {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.637834
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "my_controller"
  synchronization FALSE
  battery [
    250, 250, 30
  ]
  cpuConsumption 0.5
}
Charger {
  translation 0.9 0.9 0
  rotation 0 0 1 3.14159
  children [
    Pose {
      translation 0 0 0.15
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.20000000298023224
            roughness 0.2189099510000001
            metalness 0
            emissiveColor 0 1 0
          }
          geometry DEF CHARGER_GEOM Cylinder {
            height 0.3
            radius 0.1
            subdivision 24
          }
        }
        PointLight {
          attenuation 0 0 12
          color 0 1 0
          intensity 0.3
        }
      ]
    }
  ]
  name "charger0"
  battery [
    100, 100, 3
  ]
  radius 0.1
}
Charger {
  translation -0.9 -0.9 0
  rotation 0 0 1 -1.570795
  children [
    Pose {
      translation 0 0 0.15
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.20000000298023224
            roughness 0.2189099510000001
            metalness 0
            emissiveColor 0 1 0
          }
          geometry USE CHARGER_GEOM
        }
        PointLight {
          attenuation 0 0 12
          color 0 1 0
          intensity 0.3
        }
      ]
    }
  ]
  name "charger1"
  battery [
    100, 100, 4
  ]
  radius 0.1
}
Charger {
  translation -0.9 0.9 0
  children [
    Pose {
      translation 0 0 0.15
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.20000000298023224
            roughness 0.2189099510000001
            metalness 0
            emissiveColor 0 1 0
          }
          geometry USE CHARGER_GEOM
        }
        PointLight {
          attenuation 0 0 12
          color 0 1 0
          intensity 0.3
        }
      ]
    }
  ]
  name "charger2"
  battery [
    100, 100, 2
  ]
  radius 0.1
}
Charger {
  translation 0.9 -0.9 0
  rotation 0 0 1 1.5708
  children [
    Pose {
      translation 0 0 0.15
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.20000000298023224
            roughness 0.2189099510000001
            metalness 0
            emissiveColor 0 1 0
          }
          geometry USE CHARGER_GEOM
        }
        PointLight {
          attenuation 0 0 12
          color 0 1 0
          intensity 0.3
        }
      ]
    }
  ]
  name "charger3"
  battery [
    100, 100, 5
  ]
  radius 0.1
}
